pilot/pkg/xds/endpoint_builder.go

```diff
@@ -370,3 +386,140 @@ func buildEnvoyLbEndpoint(e *model.IstioEndpoint) *endpoint.LbEndpoint {
 
 	return ep
 }
+
+// TODO this logic is probably done elsewhere in XDS, possible code-reuse
+type mtlsChecker struct {
+	push *model.PushContext
+
+	// cache of host identifiers that have mTLS disabled
+	mtlsDisabledHosts map[string]bool
```

Comment:
nit: using `map[string]struct{}` may be preferred since it is binary (present vs not present) while this is ternary (true, false, not present)